pode/pode.py

```diff
@@ -1009,7 +1009,7 @@ def rightmost_highest(polygon: Polygon) -> Tuple[float, float]:
 def edges_on_the_right(polygon: Polygon,
                        line: LineString) -> Iterator[LineString]:
     for segment in segments(polygon.exterior):
-        if is_on_the_right(segment, line):
+        if not is_on_the_left(segment, line):

```

Comment:
collinear segments will be considered "on the right"?

Reply:
Actually, collinear segments shouldn't be accepted, since they result in an invalid ring. Also, not coinciding collinear segments will produce valid rings sometimes due to precision errors. This should be fixed. Thanks!